sdk/core/src/services/anchoringHandler.ts

```diff
@@ -0,0 +1,58 @@
+import { ELEM_DID_METHOD, JOLO_DID_METHOD } from '../_defaultConfig'
+import { RegistryApiService } from '@affinidi/internal-api-clients'
+import { DidDocumentService, KeysService } from '@affinidi/common'
+
+const getAnchoringParams = async (
+  api: RegistryApiService,
+  encryptedSeed: string,
+  password: string,
+  didDocument: any,
+  nonce: number,
+) => {
+  const keysService = new KeysService(encryptedSeed, password)
+  const { seed, didMethod } = keysService.decryptSeed()
+  if (didMethod === JOLO_DID_METHOD) {
+    const did = didDocument.id
+    const seedHex = seed.toString('hex')
+    const signedDidDocument = await keysService.signDidDocument(didDocument)
+
+    const { body: bodyDidDocument } = await api.putDocumentInIpfs({ document: signedDidDocument })
+    const didDocumentAddress = bodyDidDocument.hash
+
+    const {
+      body: { digestHex },
+    } = await api.createAnchorTransaction({ nonce, did, didDocumentAddress })
+
+    const transactionSignatureJson =
+      digestHex && digestHex !== '' ? await keysService.createTransactionSignature(digestHex, seedHex) : ''
+
+    const transactionPublicKey = KeysService.getAnchorTransactionPublicKey(seedHex, JOLO_DID_METHOD)
+    const ethereumPublicKeyHex = transactionPublicKey.toString('hex')
+
+    return { did, didDocumentAddress, ethereumPublicKeyHex, transactionSignatureJson, nonce }
+  }
+
+  if (didMethod === ELEM_DID_METHOD) {
+    const didService = DidDocumentService.createDidDocumentService(keysService)
+    const did = didService.getMyDid()
+
+    return { did, didDocumentAddress: '', ethereumPublicKeyHex: '', transactionSignatureJson: '' }
+  }
+}
+
+export const anchorDid = async (
+  api: RegistryApiService,
+  encryptedSeed: string,
+  password: string,
+  didDocument: any,
+  anchoredDidElem: boolean = false,
+  nonce: number = 0,
+): Promise<{ did: string }> => {
+  const response = await api.anchorDid({
+    ...(await getAnchoringParams(api, encryptedSeed, password, didDocument, nonce)),
```

Comment:
Potentially can fail if `getAnchoringParams` method will respond with something that is not an object or array.

Reply:
yes, `getAnchoringParams()` still have chance to return undefined if `didMethod` is not `JOLO_DID_METHOD` or `ELEM_DID_METHOD`. Better throw some expected error like `This did method is not supported` or return empty object than have runtime error here

Reply:
in case of wrong did method this function returns Promise<undefined>, resolved value is undefined, but distraction operator has identical behavior for undefined, null, {}, 0 etc.

But anyway now I prefer to add throwing error here.